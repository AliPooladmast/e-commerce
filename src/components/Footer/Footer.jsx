import {
  Facebook,
  Instagram,
  Twitter,
  WhatsApp,
  Room,
  Phone,
  MailOutline,
} from "@mui/icons-material";
import React from "react";
import style from "./Footer.module.scss";
import paymentMethods from "../../assets/images/payment-methods.jpg";
import { Link } from "react-router-dom";

function Footer() {
  return (
    <div className={style.Container}>
      <div className={style.Left}>
        <div className={style.About}>
          <h1>About Us</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio nemo
            cum, quo architecto officiis fugit ab cumque eveniet reprehenderit,
            minima voluptatem!
          </p>
        </div>
        <div className={style.SocialContainer}>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.facebook.com"
            className={style["SocialIcon--facebook"]}
          >
            <Facebook />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.instagram.com"
            className={style["SocialIcon--instagram"]}
          >
            <Instagram />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.twitter.com"
            className={style["SocialIcon--twitter"]}
          >
            <Twitter />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.whatsApp.com"
            className={style["SocialIcon--whatsapp"]}
          >
            <WhatsApp />
          </a>
        </div>
      </div>
      <div className={style.Center}>
        <h1>Usefull Links</h1>
        <ul>
          <li>
            <Link to="/" className={style.Link}>
              Home
            </Link>
          </li>
          <li>
            <Link to="/cart" className={style.Link}>
              Cart
            </Link>
          </li>
          <li>Men Fashion</li>
          <li>Women Fashion</li>
          <li>Accessories</li>
          <li>My Account</li>
        </ul>
      </div>
      <div className={style.Right}>
        <h1>Contact</h1>
        <div>
          <Room style={{ marginRight: "10px" }} />
          3949 Deer Haven Drive, Greenville, SC
        </div>
        <div>
          <Phone style={{ marginRight: "10px" }} />
          803-389-4982
        </div>
        <div>
          <MailOutline style={{ marginRight: "10px" }} />
          bexoli2205@anlubi.com
        </div>
        <img src={paymentMethods} alt="payment methods" />
      </div>
    </div>
  );
}

export default Footer;
