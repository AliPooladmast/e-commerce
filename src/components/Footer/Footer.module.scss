.Container {
  display: flex;
}

.Left {
  flex: 1;
  padding: 20px;
  display: flex;
  flex-direction: column;

  p {
    margin: 20px 0;
  }
}

.SocialContainer {
  display: flex;
}

.SocialIcon {
  height: 40px;
  width: 40px;
  color: white;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-right: 20px;

  &--facebook {
    @extend .SocialIcon;
    background-color: #3b5999;
  }

  &--instagram {
    @extend .SocialIcon;
    background-color: #e4405f;
  }

  &--twitter {
    @extend .SocialIcon;
    background-color: #55acee;
  }

  &--whatsapp {
    @extend .SocialIcon;
    background-color: #25d366;
  }
}

.Center {
  flex: 1;
  padding: 20px;

  h1 {
    font-size: 20px;
    margin-bottom: 30px;
  }

  ul {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-wrap: wrap;
  }

  li {
    width: 50%;
    margin-bottom: 10px;
  }
}

.Right {
  flex: 1;
  padding: 20px;
}
