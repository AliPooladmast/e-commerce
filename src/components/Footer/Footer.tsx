import {
  Facebook,
  Instagram,
  Twitter,
  WhatsApp,
  Room,
  Phone,
  MailOutline,
} from "@mui/icons-material";
import React from "react";
import style from "./Footer.module.scss";
import { Link } from "react-router-dom";
import { categories } from "../../data";

interface ICategory {
  id: number;
  img: string;
  imgMobile: string;
  title: string;
  category: string;
}

function Footer() {
  return (
    <div className={style.Container}>
      <div className={style.Left}>
        <div className={style.About}>
          <h1>About Us</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio nemo
            cum, quo architecto officiis fugit ab cumque eveniet reprehenderit,
            minima voluptatem!
          </p>
        </div>
        <div className={style.SocialContainer}>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.facebook.com"
            className={style["SocialIcon--facebook"]}
          >
            <Facebook />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.instagram.com"
            className={style["SocialIcon--instagram"]}
          >
            <Instagram />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.twitter.com"
            className={style["SocialIcon--twitter"]}
          >
            <Twitter />
          </a>
          <a
            target="_blank"
            rel="noreferrer"
            href="https://www.whatsApp.com"
            className={style["SocialIcon--whatsapp"]}
          >
            <WhatsApp />
          </a>
        </div>
      </div>
      <div className={style.Center}>
        <h1>Usefull Links</h1>
        <ul>
          <li>
            <Link to="/" className={style.Link}>
              Home
            </Link>
          </li>
          <li>
            <Link to="/cart" className={style.Link}>
              Cart
            </Link>
          </li>

          {categories.map((item: ICategory) => (
            <li key={item.id}>
              <Link to={`/products/${item.category}`} className={style.Link}>
                {item.title}
              </Link>
            </li>
          ))}

          <li>
            <Link to="/user" className={style.Link}>
              My Account
            </Link>
          </li>
        </ul>
      </div>
      <div className={style.Right}>
        <h1>Contact</h1>
        <div>
          <Room style={{ marginRight: "10px" }} />
          3949 Deer Haven Drive, Greenville, SC
        </div>
        <div>
          <Phone style={{ marginRight: "10px" }} />
          803-389-4982
        </div>
        <div>
          <MailOutline style={{ marginRight: "10px" }} />
          bexoli2205@anlubi.com
        </div>
        <img src="assets/images/payment-methods.jpg" alt="payment methods" />
      </div>
    </div>
  );
}

export default Footer;
