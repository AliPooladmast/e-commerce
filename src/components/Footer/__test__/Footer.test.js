const { render, screen } = require("@testing-library/react");
const { BrowserRouter } = require("react-router-dom");
import Footer from "../Footer";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

describe("links validity", () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <Footer />
      </BrowserRouter>
    );
  });

  it("check Home link availability", () => {
    expect(screen.getByText(/home/i)).toBeInTheDocument();
  });

  it("check Cart link availability", () => {
    expect(screen.getByText(/cart/i).href).toMatch(/cart/);
  });

  it("check User link availability", () => {
    expect(screen.getByText(/My Account/i).href).toMatch(/user/);
  });

  it("check casual category link availability", () => {
    expect(
      screen.getByText(/casual/i).href.includes("/products/casual")
    ).toBeTruthy();
  });

  it("check formal category link availability", () => {
    expect(
      screen.getByText(/formal/i).href.includes("/products/formal")
    ).toBeTruthy();
  });

  it("check fashion category link availability", () => {
    expect(
      screen.getByText(/fashion/i).href.includes("/products/fashion")
    ).toBeTruthy();
  });
});
