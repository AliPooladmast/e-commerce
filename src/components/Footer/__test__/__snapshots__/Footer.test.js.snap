// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Container"
>
  <div
    className="Left"
  >
    <div
      className="About"
    >
      <h1>
        About Us
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio nemo cum, quo architecto officiis fugit ab cumque eveniet reprehenderit, minima voluptatem!
      </p>
    </div>
    <div
      className="SocialContainer"
    >
      <a
        className="SocialIcon--facebook"
        href="https://www.facebook.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="FacebookIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
          />
        </svg>
      </a>
      <a
        className="SocialIcon--instagram"
        href="https://www.instagram.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="InstagramIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
          />
        </svg>
      </a>
      <a
        className="SocialIcon--twitter"
        href="https://www.twitter.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </a>
      <a
        className="SocialIcon--whatsapp"
        href="https://www.whatsApp.com"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WhatsAppIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.75 13.96c.25.13.41.2.46.3.06.11.04.61-.21 1.18-.2.56-1.24 1.1-1.7 1.12-.46.02-.47.36-2.96-.73-2.49-1.09-3.99-3.75-4.11-3.92-.12-.17-.96-1.38-.92-2.61.05-1.22.69-1.8.95-2.04.24-.26.51-.29.68-.26h.47c.15 0 .36-.06.55.45l.69 1.87c.06.13.1.28.01.44l-.27.41-.39.42c-.12.12-.26.25-.12.5.12.26.62 1.09 1.32 1.78.91.88 1.71 1.17 1.95 1.3.24.14.39.12.54-.04l.81-.94c.19-.25.35-.19.58-.11l1.67.88M12 2a10 10 0 0 1 10 10 10 10 0 0 1-10 10c-1.97 0-3.8-.57-5.35-1.55L2 22l1.55-4.65A9.969 9.969 0 0 1 2 12 10 10 0 0 1 12 2m0 2a8 8 0 0 0-8 8c0 1.72.54 3.31 1.46 4.61L4.5 19.5l2.89-.96A7.95 7.95 0 0 0 12 20a8 8 0 0 0 8-8 8 8 0 0 0-8-8z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="Center"
  >
    <h1>
      Usefull Links
    </h1>
    <ul>
      <li>
        <a
          className="Link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/cart"
          onClick={[Function]}
        >
          Cart
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/products/casual"
          onClick={[Function]}
        >
          Casual Outfit
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/products/fashion"
          onClick={[Function]}
        >
          Fashion Style
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/products/formal"
          onClick={[Function]}
        >
          Formal Dress
        </a>
      </li>
      <li>
        <a
          className="Link"
          href="/user"
          onClick={[Function]}
        >
          My Account
        </a>
      </li>
    </ul>
  </div>
  <div
    className="Right"
  >
    <h1>
      Contact
    </h1>
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="RoomIcon"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
      3949 Deer Haven Drive, Greenville, SC
    </div>
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="PhoneIcon"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
        />
      </svg>
      803-389-4982
    </div>
    <div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MailOutlineIcon"
        focusable="false"
        style={
          Object {
            "marginRight": "10px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
        />
      </svg>
      bexoli2205@anlubi.com
    </div>
    <img
      alt="payment methods"
      src="assets/images/payment-methods.jpg"
    />
  </div>
</div>
`;
